### PR TITLE
3671: Create a "SetPaketCommand" target to be able to try Exec the local cli tool

### PR DESCRIPTION
**Pull request by atlemann**
Suggested by @NatElkins.

Fixes: #3671

### CODE REVIEW

File: src/Paket.Core/embedded/Paket.Restore.targets [hunk 1]
```diff
@@ -75,14 +67,38 @@
...
+      <Output TaskParameter="ExitCode" PropertyName="LocalPaketToolExitCode" />
+    </Exec>
+
+    <!-- Then we go through our normal steps of setting the Paket command.
+         I called it InternalPaketCommand here because apparently properties aren't available
+         outside the targets they are declared in. -->
+    <PropertyGroup>
+      <_PaketExeExtension>$([System.IO.Path]::GetExtension("$(PaketExePath)"))</_PaketExeExtension>
+      <InternalPaketCommand Condition=" '$(InternalPaketCommand)' == '' AND '$(_PaketExeExtension)' == '.dll' ">dotnet "$(PaketExePath)"</InternalPaketCommand>
+      <InternalPaketCommand Condition=" '$(InternalPaketCommand)' == '' AND '$(_PaketExeExtension)' == '' AND $(LocalPaketToolExitCode) == 0">dotnet "$(PaketExePath)"</InternalPaketCommand>
```
Comment (atlemann) [inline src/Paket.Core/embedded/Paket.Restore.targets]: I guess we could start with checking if `LocalPaketToolExitCode == 0` and if it is, just set `InternalPaketCommand` to `dotnet paket`? @NatElkins

**Comment by atlemann** [comment]
We could consider this way instead as mentioned in the issue:

```xml
<PropertyGroup>
  <DotnetToolsJson>($([System.IO.File]::ReadAllText("$(PaketRootPath)/.config/dotnet-tools.json")))</DotnetToolsJson>
  <ContainsPaket>$(DotnetToolsJson.Contains('"paket"'))</ContainsPaket>
</PropertyGroup>
```

**Comment by forki** [comment]
Oh CI is red?

**Comment by atlemann** [comment]
Yes, couldn't see why.

**Comment by forki** [comment]
Let's rerun this

**Comment by atlemann** [comment]
Ok, the first commit passed and I only moved the paketexe search stuff into the target in the second one to avoid doing that work if not used. But maybe I missed a usage somewhere.

**Comment by atlemann** [comment]
Passed now

File: src/Paket.Core/embedded/Paket.Restore.targets (hunk 1 above)
Comment (NatElkins) [reply src/Paket.Core/embedded/Paket.Restore.targets]: Sure whatever you think is clearest to someone coming back a year from now to try and decipher this.

**Comment by forki** [comment]
Thanks so much for the help!

**Pull request merged**